src/main.c

```diff
@@ -526,12 +526,12 @@ static void IRQ_osd_end(void)
     if (startup_display_spi == DISP_SPI1) {
         dma_display_spi1.ccr = 0;
         dma_display_spi1.cndtr = cur_display->cols/2 + 1;
-        if ((config.display_2Y == FALSE) || (hline & 0x1))
+        if ((config.display_2Y == FALSE) || ((vstart + hline) & 0x1))
```

Comment:
This makes no sense, hline and vstart are both relative to top of screen. I think you mean `hline - vstart`, or some variant of that. As what you care about is vertical offset from the top of the OSD. And that's what `vstart` is.

Reply:
Mathematically it does but only because I’m looking for even/odd.
Ie, is this the first or second of a duplicated row.
Yours will work too - I’ll change it and add a comment on the logic here

Reply:
Oh yes I see, since `vstart+hline == (hline-vstart)+2*vstart`. Well, it's still confusing ;)